pydicer/input/tcia.py

```diff
@@ -0,0 +1,93 @@
+import os
+import urllib.request, urllib.error, urllib.parse
+import urllib.request, urllib.parse, urllib.error
+from pydicer.input.base import InputBase
+import zipfile
+from pathlib import Path
+
+#
+# Refer https://wiki.cancerimagingarchive.net/display/Public/REST+API+Usage+Guide for complete
+# list of API
+#
+# Some of the code was adapted from the TCIA Python APi client example:
+# https://github.com/TCIA-Community/TCIA-API-SDK/blob/master/tcia-rest-client-python/src/tciaclient.py
+#
+
+
+def run_endpoint(url, query_parameters):
+    """
+    Execute the TCIA API endpoint
+
+    Args:
+        url (str): url of the series to be downloaded and generate its DICOMS
+        query_parameters (dict): dictionary of query parameters to be passed into the api
+        endpoint request.
+
+    Returns:
+        resp (file): the response file in bytes to be written into a zipfile
+    """
+
+    query_parameters = dict((k, v) for k, v in query_parameters.items() if v)
+    queryString = "?%s" % urllib.parse.urlencode(query_parameters)
+    requestUrl = url + queryString
+    request = urllib.request.Request(url=requestUrl)
+    resp = urllib.request.urlopen(request)
+    return resp
+
+
+class TCIAInput(InputBase):
+    def __init__(self, series_instance_uid, working_directory=None):
```

Comment:
Rather than specifying the series instance UID here, can we specify a TCIA collection, and a single or list of patients to fetch from the collection? An additional option could be the modalities that we are interested in.

Reply:
Since in pydicer we generally are looking at grabbing all of the data for a patient, since we don't yet know exactly what we want to use from it until after we have converted to Nifti etc... :)

Reply:
Yeah for sure, I was unsure at first how well-defined a user wanted to go with the api query so I chose the lowest form as a start, but makes sense to go by collection/patient/modality(opt) as a complete form, easy change

Reply:
Ok turns out the platipy tcia function already does this haha. I've re-written my implementation of it to include collection/patient/modality and tested both platipy's and mine to see which is faster for a range of inputs. There is any significant difference between the 2 and Platipy handles the cases we want already. So I think we should just revert to using platipy's tcia.fetch_data()